django/db/backends/mysql/schema.py

```diff
@@ -70,12 +70,22 @@ def _is_limited_data_type(self, field):
         )
 
     def skip_default(self, field):
+        is_limited_data_type = self._is_limited_data_type(field)
+        default_not_provided = field.default is NOT_PROVIDED
+        default_is_empty = self.effective_default(field) in ("", b"")
+        if is_limited_data_type and default_not_provided and default_is_empty:
```

Comment:
I'd like to see us make sure that we only take this path for `blob` and `text` fields. The way it's written now would cause this path to be taken for `json` columns as well as any future addition to `_limited_data_types` https://github.com/django/django/blob/e65d6adf3cdae2cae30f01b93e4ded3007c9297f/django/db/backends/mysql/base.py#L149-L159.

Maybe we could add an extra dict attribute that tracks the implicit default associated with `columns` e..g

```
_columns_implicit_default = (
    "tinyblob": b"",
    "blob": b"",
    "mediumblob": b"",
    "longblob": b"",
    "tinytext": "",
    "text": "",
    "mediumtext": "",
    "longtext": "",
    "int": 0,
    ...
    "json": "null",
    ...
)
```

And use this mapping instead of special casing `("", b"")` and values.

Reply:
Something like

```python
def _get_implicit_default(self, field):
    db_type = field.db_type(self.connection)
    if db_type is None:
        return None
    return self.connection._data_types_implicit_default.get(db_type.lower())
```

And then

```python
def skip_default(self, field):
    if self._get_implicit_default(field) == self.effective_default(field):
        return True
```